File: JavaScript/MaximumAverageSubarrayI/MaximumAverageSubarrayI.js
/**
 * (168ms)
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    var now = 0;
    for(let i = 0; i < k; i++) {
    	now += nums[i];
    }
    var max = now, head = k, tail = 0;
    while(head < nums.length) {
    	now = now - nums[tail++] + nums[head++];
    	max = Math.max(max, now);
    }
    return max / k;
};

File: Python/MaximumAverageSubarrayI/MaximumAverageSubarrayI.py
class Solution(object):
	# (255ms)
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        now = sum(nums[:k])
        head, tail, maxSum = k, 0, now
        while head < len(nums):
        	now = now - nums[tail] + nums[head]
        	maxSum = max(maxSum, now)
        	tail += 1
        	head += 1
        return maxSum / k
